### PR TITLE
added search functionality to voice path

### PR DESCRIPTION
**Pull request by alphaseinor**
# Description
we discussed a way to search tags (or attributes), this is one implementation of a "live search style"

Fixes # (issue)
feature request

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: voicer-client/src/components/voice/Voice.jsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import VoiceItem from "./VoiceItem";
 export default function Voice() {
   const [nameMatchesDB, setNameMatchesDB] = useState(true);
   const [data, setData] = useState([]);
+  const [voiceSearch, setVoiceSearch] = useState('')
+  const [searchTags, setSearchTags] = useState('')
```
Comment (davidjayfrancis) [inline voicer-client/src/components/voice/Voice.jsx:11]: @alphaseinor Wouldn't tags be an array given there are multiple possible tags that can be added?

File: voicer-client/src/components/voice/Voice.jsx [hunk 2]
```diff
@@ -32,12 +34,13 @@ export default function Voice() {
           setNameMatchesDB(true);
         });
     } else {
-      //console.log("there is no param")
+      //parse search string into multiple search items
+      //console.log(`sending to: ${url}/api/users${searchTags}`)
       axios
-        .get(`${url}/api/users`)
+        .get(`${url}/api/users${searchTags}`)
```
Comment (davidjayfrancis) [inline voicer-client/src/components/voice/Voice.jsx:40]: So the backend will receive something like "TAG1 TAG2 TAG3"?

File: voicer-client/src/components/voice/Voice.jsx (hunk 1 above)
Comment (alphaseinor) [reply voicer-client/src/components/voice/Voice.jsx:11]: this is so you can have a search bar that doesn't include things like ?tag0=this&tag1=is as you type in it. you need a change handler and something to parse the search query into the get request

File: voicer-client/src/components/voice/Voice.jsx (hunk 2 above)
Comment (alphaseinor) [reply voicer-client/src/components/voice/Voice.jsx:40]: ?tag0=this&tag1=is

File: voicer-client/src/components/voice/Voice.jsx (hunk 1 above)
Comment (davidjayfrancis) [reply voicer-client/src/components/voice/Voice.jsx:11]: Makes sense. One thing we might do in a future release is make it so when a user presses space it auto-completes a tag (even if just styling on frontend) to make more obvious to the user that this is a tag search.

**Pull request merged**